Pool/HighScores.vb

```diff
@@ -30,19 +32,73 @@ Public Class HighScores
         allGames = games.GetAllPlayers()
         highScoreTheme.FillCBoxAll(allGames, cbGames, lblError)
         highScores = games.GetAllResults("exec selAllScores @output=0")
-
         lblError.Visible = False
         If highScoreTheme.Screen = -1 Then
             lblError.Visible = True
         Else
             highScoreTheme.Screen = ScoreTheme.AppState.SelectPlayer
         End If
+        ProcessEmailStatbyWeek()
+#Region "control buttons loc/dimens"
         addLoc = btnAdd.Location
         saveLoc = btnSubmit.Location
         saveDim = btnSubmit.Size
         addDim = btnAdd.Size
+#End Region
+    End Sub
+
+    ''' <summary>
+    ''' Gets a list of recipents from db
+    ''' Get a stats from recent stats the past week without timestamps
+    ''' Pass in the email obj and call send function
+    ''' Using the stats just grabed, get their IDs and update timestamps
+    ''' </summary>
+    Private Sub ProcessEmailStatbyWeek()
+        Try
+            Dim emailRecipents As New List(Of String)
+            emailRecipents = highScoreTheme.getSubs(games)
+            Dim recentStats As IEnumerable = GetRecentStats(highScores.Tables(0))
+            Dim emailSet As New Email(emailRecipents, recentStats)
+            emailSet.SendWeekEmail()
+            Dim sentID As String = GetStatsSent(recentStats)
+            If sentID.Length > 0 Then
+                games.InsertGame($"exec [updEmailSent] @hid='{sentID}',@timesent='{Now.ToString("")}'")
+            End If
+        Catch ex As Exception
+            Debug.Write(ex.ToString)
+        End Try
     End Sub
 
+    ''' <summary>
+    ''' Filter the stats by the lastest in the past week
+    ''' </summary>
+    ''' <param name="allStats"></param>
+    ''' <returns></returns>
+    Public Function GetRecentStats(allStats As DataTable) As IEnumerable
+        Try
+            Dim recentScore As IEnumerable = (From row As DataRow In allStats.AsEnumerable
+                                              Where DateDiff(DateInterval.Day, row.Field(Of Date)("lastUpdated"), Now) <= 7 AndAlso
+                                                   row.Field(Of Date?)("lastSent").ToString.Count <= 0
+                                              Order By row.Field(Of Integer)("highscore") Descending).ToArray
```

Comment:
Make sure this will pull only all results under these:

1. The timestamp is within a week from the time the application is run for now.
2. That when checking the "lastSent" field, if there is no/null we are not including it (if considered empty)or won't cause the query to throw an exception.

Reply:
Looks good during testing.